test/puppetlabs/trapperkeeper/services/metrics/metrics_service_test.clj

```diff
@@ -61,4 +61,26 @@
                             "pl.foo.reg:name=puppetlabs.localhost.foo")))
                 body (parse-response resp)]
             (is (= 200 (:status resp)))
-            (is (= {"Value" 2} body))))))))
+            (is (= {"Value" 2} body)))))
+
+
+      (testing "querying multiple metrics via POST should work"
+        (let [svc (app/get-service app :MetricsService)
+              registry (metrics-protocol/get-metrics-registry svc
+                                                              ::my-other-reg
+                                                              "pl.other.reg")]
+          (metrics/register registry
+                            (metrics/host-metric-name "localhost" "foo")
+                            (metrics/gauge 2))
+          (metrics/register registry
+                            (metrics/host-metric-name "localhost" "bar")
+                            (metrics/gauge 500))
+
+          (let [resp (http-client/post
+                      "http://localhost:8180/metrics/v1/mbeans"
+                      {:body (json/generate-string
+                              ["pl.other.reg:name=puppetlabs.localhost.foo"
+                               "pl.other.reg:name=puppetlabs.localhost.bar"])})
+                body (parse-response resp)]
+            (is (= 200 (:status resp)))
```

Comment:
Pushed up tests for the other cases and added the ability to post a string or object as well :) @fpringvaldsen